pymc3/distributions/distribution.py

```diff
@@ -147,18 +147,20 @@ def default(self):
 
     def get_test_val(self, val, defaults):
         if val is None:
-            for v in defaults:
-                if hasattr(self, v) and np.all(np.isfinite(self.getattr_value(v))):
-                    return self.getattr_value(v)
-        else:
-            return self.getattr_value(val)
-
-        if val is None:
+            try:
+                for v in defaults:
+                    if hasattr(self, v) and np.all(np.isfinite(self.getattr_value(v))):
```

Comment:
No point in calling `self.getattr_value` twice specially since most cases are probably not going to be infinite. Would you like to make that change?

Reply:
This would require adding one more `if` case cause we'll need the `hasattr(self, v)` evaluated to `True` check before we can actually do `self.getattr_value(v)`. I think it's better to just let it be. I don't think it'll make much of a difference but the code looks cleaner this way rather than a bunch of `if` cases.